extensions-contrib/virtual-columns/src/main/java/io/druid/segment/MapVirtualColumn.java

```diff
@@ -115,6 +121,12 @@ public Map getObject()
       }
       return new MapVirtualColumnValueSelector<String>(keySelector, valueSelector)
       {
+        @Override
+        public boolean isNull()
```

Comment:
Maybe this method should return enum NULL, NOT_NULL, UNKNOWN, where the third option literally means calling `getObject()` and compare it with null. Because currently it fosters performing duplicate value computation sometimes.

If `isNull()` is called in some hot loops (I suppose), I suggest to make optimization from the beginning and use C-like int codes 0, 1, -1 instead of proper java enum. (Because enum comparison will compare with addresses, while 0, 1, -1 will produce almost or exactly as efficient code as `boolean`.)

Reply:
In hot loops we can directly call getObject and compare that with null and avoid calling isNull.

Reply:
Yes, but in this case `getObject()` is far from being cheap. In some other cases you cannot call getObject() if `isNull()` returns `true`, it will lead to exception or simply wrong return.

Reply:
Please address

Reply:
would prefer to do it in a separate PR - https://github.com/druid-io/druid/issues/5169. 
Hope its not a blocker since SQL-null handling is disabled by default.

Reply:
Also, default implementation of `AggregateCombiner.isNull()` and `Aggregator.isNull()` is quite misleading (see caveat in the doc of the latter), but with three-value, it could be UNKNOWN that's better I think. And NOT_NULL for primitive-valued AggregateCombiners and Aggregators.